modules/compiler/vecz/source/transform/packetizer.cpp

```diff
@@ -1589,6 +1622,244 @@ Packetizer::Result Packetizer::Impl::packetizeSubgroupShuffleXor(
   return assign(CI, CombinedShuffle);
 }
 
+Packetizer::Result Packetizer::Impl::packetizeSubgroupShuffleUpDown(
+    Instruction *I, compiler::utils::GroupCollective ShuffleUpDown) {
+  bool IsDown =
+      ShuffleUpDown.Op == compiler::utils::GroupCollective::OpKind::ShuffleDown;
+  assert((IsDown || ShuffleUpDown.Op ==
+                        compiler::utils::GroupCollective::OpKind::ShuffleUp) &&
+         "Invalid shuffle kind");
+
+  auto *const CI = cast<CallInst>(I);
+
+  // We don't support scalable vectorization of sub-group shuffles.
+  if (SimdWidth.isScalable()) {
+    return Packetizer::Result(*this);
+  }
+  unsigned const VF = SimdWidth.getFixedValue();
```

Comment:
I'd personally prefer something more decriptive - I was looking at code below and thinking what's VF?

Reply:
Ah, I thought it was well-established by this point that it means "vectorization factor". That's why we have `CA_RISCV_VF`. We use that terminology in `WorkItemLoopsPass`

It's admittedly a bit inconsistent, but I personally have a half-plan to rename `SimdWidth` to something closer to `VF` across the entire vectorizer.

Reply:
It is subtly different to what we call the vectorization factor, as it's only SimdWidth, but feel free to ignore the suggestion.